### PR TITLE
feat(Icon): add pin solid icon

### PR DESCRIPTION
**Pull request by sachalifs**
[PRO-2299](https://toptal-core.atlassian.net/browse/PRO-2299)

### Description

We need this pin solid icon for the Talent's Internal Resume Page initiative.

This icon is part of [our BASE design system](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/c9ead95be2056da8ab40cd4bc5d10b01525d1ceb/files/2fbda9fd-63f3-42e7-a9a8-135a6562ac01/layers/00EEBF49-F0F0-454C-B616-F744B8796629).

image

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

File: packages/picasso/src/Icon/svg/pinSolid16.svg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            </mask>
+            <use id="Combined-Shape" fill="#000000" fill-rule="nonzero" xlink:href="#path-1"></use>
+            <g id="Group" mask="url(#mask-2)" fill="#455065">
+                <g id="Colors-/-307-Gray-Dark">
+                    <rect id="Rectangle" x="0" y="0" width="16" height="16"></rect>
+                </g>
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (dmaklygin1) [inline packages/picasso/src/Icon/svg/pinSolid16.svg:20]: question: shouldn't we pass the SVG code through the optimizer like SVGO?

Comment (dmaklygin1) [reply packages/picasso/src/Icon/svg/pinSolid16.svg:20]:
```
<svg width="16" height="16" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><path d="M8 0a6 6 0 016 6c0 2.21-2 5.542-6 10-4-4.458-6-7.79-6-10a6 6 0 016-6zm0 4a2 2 0 100 4 2 2 0 000-4z" id="a"/></defs><g fill="none" fill-rule="evenodd"><mask id="b" fill="#fff"><use xlink:href="#a"/></mask><use fill="#000" fill-rule="nonzero" xlink:href="#a"/><g mask="url(#b)" fill="#455065"><path d="M0 0h16v16H0z"/></g></g></svg>
```

Comment (OleksandrNechai) [reply packages/picasso/src/Icon/svg/pinSolid16.svg:20]: Looks like most raw SVG files are not compressed and compression is happening during `tsx` file generation though.

**Comment by OleksandrNechai** [comment]
@toptal-bot run all

**Review by dmaklygin1 — APPROVED** [review]

**Pull request merged**